### DISCUSSION TITLE
Multibooking form multi-language not displaying properly

### DISCUSSION THREAD

**Discussion by refugeindhamma** [liedekef/events-made-easy#59] [Q&A] [answered]
Hi Franky,

Multi-language tags in multibooking form body & footer templates not working properly for Chinese.
![Screenshot 2023-02-03 at 17-39-18 忆佛念混合禅修课程注册 2023年 3月](https://user-images.githubusercontent.com/101639308/216567413-b8f9b93a-f9a6-4263-8115-d61ef997e051.png)

In the screenshot above, it can be seen that the header and body are working properly.
~~But for the body, there are English words mixed in the Chinese form like "Please select in-person".~~ (I realized the English words were included in the Chinese translation by mistake.)
For the footer, there is a duplication of GDPR and some words are missing.

You can see on this screenshot that even the English site is displaying duplicate GDPR:
![Screenshot 2023-02-03 at 17-52-33 Recollection of the Buddha Meditation Course Registration Mar 2023](https://user-images.githubusercontent.com/101639308/216569392-ef5d08de-3f25-4446-b7cb-a97b53f25ac0.png)

And I do not know why it is showing refusal of newsletter.

Multibooking Header Template:
![Screenshot 2023-02-03 at 18-53-37 Templates ‹ Refuge In Dhamma — WordPress](https://user-images.githubusercontent.com/101639308/216584987-2bcf5844-777e-4fa5-9166-f8a4a789406c.png)

Multibooking Body Template:
![Screenshot 2023-02-03 at 18-49-42 Templates ‹ Refuge In Dhamma — WordPress](https://user-images.githubusercontent.com/101639308/216585206-84551f7b-3758-42e3-afd2-f908dcfe03e7.png)

Multibooking Footer Template:
![Screenshot 2023-02-03 at 18-50-12 Templates ‹ Refuge In Dhamma — WordPress](https://user-images.githubusercontent.com/101639308/216585066-399f2918-ebd7-4e47-bf9c-5a6f09ff97da.png)

How do I resolve this?

**Comment by refugeindhamma** [comment]
Another matter that is out-of-topic, how do I put code in github that does not mess up the formatting?

When I tried to put the above codes into github using the code mark-down, it displays the table for the header and the body code just messes up the formatting.

**Reply by liedekef** [reply]
github itself uses markdown, best to use three backticks for fixed format.

**Comment by liedekef** [comment]
Your footer template contains an error: you have '`</ul>`' above the [:en] tag, but also below the [:zh] tag ...
The rest: I'll check but currently don't see anything wrong in the code. Do try to upload your format-code directly, so I can copy/paste :-)

**Reply by refugeindhamma** [reply]
I corrected the error and put the language tag above </ul> but it's still the same.

**Reply by refugeindhamma** [reply]
Header Template:
```
[:en]
<table style="width: 100%; border-collapse: collapse;" border="0" cellpadding="20">
<tbody>
<tr>
<td style="width: 50%;"><strong>Surname<span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">*</span></strong><br />#_LASTNAME{Eg: Tan}</td>
<td style="width: 50%;"><strong>Given Name<span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">*</span></strong><br />#_FIRSTNAME{Eg: Ah Meng}</td>
</tr>
<tr>
<td style="width: 50%;"><strong>Email<span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">*</span></strong><br />#REQ_EMAIL{Eg: tanahmeng@email.com}</td>
<td style="width: 50%;"><strong>Handphone<span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">*</span></strong><br />#REQ_PHONE{+65 12345678}</td>
</tr>
<tr>
<td style="width: 50%;"><strong>Country<span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">*</span></strong><br />#_COUNTRY</td>
<td style="width: 50%;">&nbsp;</td>
</tr>
</tbody>
</table>
<BR />
<strong>Please select In-person <span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">OR</span> Virtual for the following dates:</strong>
<BR /><BR />
<ul>
[:zh]
<table style="width: 100%; border-collapse: collapse;" border="0" cellpadding="20">
<tbody>
<tr>
<td style="width: 50%;"><strong>姓氏<span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">*</span></strong><br />#_LASTNAME{如：Tan（陈）}</td>
<td style="width: 50%;"><strong>名字<span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">*</span></strong><br />#_FIRSTNAME{如：Ah Meng（阿明）}</td>
</tr>
<tr>
<td style="width: 50%;"><strong>电子邮件<span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">*</span></strong><br />#REQ_EMAIL{如：tanahmeng@email.com}</td>
<td style="width: 50%;"><strong>手机<span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">*</span></strong><br />#REQ_PHONE{+65 12345678}</td>
</tr>
<tr>
<td style="width: 50%;"><strong>国家<span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">*</span></strong><br />#_COUNTRY</td>
<td style="width: 50%;">&nbsp;</td>
</tr>
</tbody>
</table>
<BR />
<strong>请为以下的日期选择当场Please select In-person <span style="color: #ff0000; --darkreader-inline-color: #ff1a1a;" data-darkreader-inline-color="">或者</span> 线上：</strong>
<BR /><BR />
<ul>
[:]
```

Body Template:
```
[:en]
<li>
<em>#_EVENTNAME</em>
<BR />
#_{d M Y (l)} #_24HSTARTTIME - #_24HENDTIME #_SEATS
<BR /><BR />
</li>
[:zh]
<li>
<em>#_EVENTNAME</em>
<BR />
#_{Y年 M d日 (l)} #_24HSTARTTIME - #_24HENDTIME 时 #_SEATS
<BR /><BR />
</li>
[:]
```

Footer Template:
```
[:en]
</ul>
<br />
<strong>Newsletter<span style="color:red">*</span>:</strong>&nbsp;#REQ_OPT_OUT<br />
<br />
#_GDPR I have read and agree to the <a href="/terms-and-conditions" rel="noopener" target="_blank">Terms & Conditions</a>. I allow RID to use my personal information according to the <a href="/privacy-policy" rel="noopener" target="_blank">Privacy Policy</a>.<br />
<br />
#_CFCAPTCHA
#_SUBMIT<br />
[:zh]
</ul>
<br />
<strong>信息通讯<span style="color:red">*</span>:</strong>&nbsp;#REQ_OPT_OUT<br />
<br />
#_GDPR 我已阅读并同意该<a href="/条款与条件" rel="noopener" target="_blank">条款与条件</a>。我允许RID根据<a href="/隐私政策" rel="noopener" target="_blank">隐私政策</a>使用我的个人信息。<br />
<br />
#_CFCAPTCHA
#_SUBMIT{注册}<br />
[:]
```

**Comment by liedekef** [comment]
In your English footer section, you ask for newsletter opt_out: `<strong>Newsletter<span style="color:red">*</span>:</strong>&nbsp;#REQ_OPT_OUT<br />` , not in the Chinese version. I don't see anything else wrong there, all concerning header/footer works as expected for en en zh.
I'm now checking the template concerning the "body"

**Reply by refugeindhamma** [reply]
It's there just above GDPR.
```
<strong>信息通讯<span style="color:red">*</span>:</strong>&nbsp;#REQ_OPT_OUT<br />
```

**Comment by liedekef** [comment]
I don't see anything wrong in the body either, the only thing I see is your date format being "Y年 M d日 (l)" ==> check the php datetime formatting options to see if M, d are really the options you want.

**Reply by refugeindhamma** [reply]
Yup, that's the date format I want. However, it's strange that d Y is not translated to Chinese form of Y年 d日. Only the month is translated correctly. That's why I had to put in the chinese characters for year & day in the template.

**Comment by refugeindhamma** [comment]
The only CSS I have added to the page is this:
```
.registration ul li{
    float: left;
    width: 50%;//helps to determine number of columns, for instance 33.3% displays 3 columns
}
ul{
    list-style-type: disc;
}
```
I have tried removing the code and the page still displays duplicate EN & ZH GDPR and register buttons.

**Reply by refugeindhamma** [reply]
If I separate the EN & ZH templates, everything works fine.
Edit: ~~I'm using Kadence Theme + Kadence Blocks. Not sure if it is Kadence causing this to happen.~~ (Turned off Kadence Theme + Blocks and the duplication is still there)
![image](https://user-images.githubusercontent.com/101639308/216742132-620c2ad1-fd6b-45f6-ac7d-4bf192fd642a.png)

**Reply by refugeindhamma** [reply]
When there is such a duplication, I need to check both GDPR boxes in order to submit the form or else there will be an error saying that I did not check the GDPR box if only either one of the boxes are checked.

**Comment by refugeindhamma** [comment]
Currently, separating the language files, i.e. one template containing only one language, for footer template works but to be safe, I separated all 3 templates: header, body & footer.

**Comment by liedekef** [comment]
can you mail me an example of an event on your dev-site that has this issue?

**Reply by refugeindhamma** [reply]
It is actually the same event that I sent to you on Issue #60.

Edit: The [English site](https://dev.refugeindhamma.com/recollection-of-the-buddha-meditation-course-registration-mar-2023/) shows duplicate GDPR. This is the [Chinese site](https://dev.refugeindhamma.com/zh/%e5%bf%86%e4%bd%9b%e5%bf%b5%e6%b7%b7%e5%90%88%e7%a6%85%e4%bf%ae%e8%af%be%e7%a8%8b%e6%b3%a8%e5%86%8c2023%e5%b9%b43%e6%9c%88/) with a separate Chinese template for the footer and it is working correctly.

**Comment by liedekef** [comment, marked as answer]
Ok, I figured it out: it is because due to placeholder expansion, language tags get added "inside" other language tags, causing this issue. The language tag code is not recursive :-)
I've now corrected this by - at the moment the template is get - I translate the template body already, solving this issue (check your dev-site).

**Comment by refugeindhamma** [comment]
Great work Franky! Thanks for resolving all the Multibooking issues so quickly. Really appreciate it. Gave you a small donation.

**Reply by liedekef** [reply]
The code is not released yet, so you'll need to wait a couple of days for the prod-release :-)
